Update ReadMe For tree (#21691)
microsoft/FluidFramework: packages/dds/tree/README.md

Notable consideration that early adopters should be wary of:

-   The persisted format is not stable and long term support for persisted format is not yet in effect.
    This means clients running more recent versions of SharedTree may find themselves unable to load old document or may corrupt them when editing.
    Current estimate for LTS is April 2024.
-   SharedTree currently has unbounded memory growth:

>     -   Removed content is retained forever.

Reviewer: Until [https://github.com/microsoft/FluidFramework/pull/21372/](https://github.com/microsoft/FluidFramework/pull/21372/) is merged, removed content will keep on being retained forever.

Author: I have updated the text to mention this.

> -   The persisted format is now stable: documents created with released versions 2.0.0 of greater of this package will be supported long term.

Reviewer 2: ```suggestion
-   The persisted format is stable such that documents created with released versions 2.0.0 or greater of this package are fully supported long term.
```

> -   Some documentation (such as this readme and [the roadmap](docs/roadmap.md)) are out of date.

Reviewer 2: If this is correct, surely we should update this readme, and not say it is out of date?

Author: there are plenty of out of date things further below.

Notable consideration that early adopters should be wary of:

-   The persisted format is stable such that documents created with released versions 2.0.0 or greater of this package are fully supported long term.
-   SharedTree currently has unbounded memory growth:

>     -   Removed content is retained in memory for the duration of the session ([fix](https://github.com/microsoft/FluidFramework/pull/21372)).

Reviewer: ```suggestion
    -   Removed content is retained in memory and persisted in the document at rest ([fix](https://github.com/microsoft/FluidFramework/pull/21372)).
```
